File: Export/SupabaseExporter/SupabaseExporter/Structures/Exports/Coffer.cs
namespace SupabaseExporter.Structures.Exports;

/// <summary>
/// The base of coffer data.
/// </summary>
[Serializable]
public class Coffer(string name, uint territoryId, List<Coffer.Variant> variants)
{
    public string Name = name;
    public uint TerritoryId = territoryId;
    public List<Variant> Variants = variants;

    public record Variant(uint Id, string Name, Dictionary<string, Content> Patches);
    public record Content(long Total, List<Reward> Items);
}

File: website/src/lib/structs/coffer.ts
import type {Reward} from "$lib/structs/reward";

export interface Coffer {
    Name: string;
    TerritoryId: number;

    Variants: CofferVariant[];
}

export interface CofferVariant {
    Id: number;
    Name: string;

    Patches: Record<string, CofferContent>;
}

export interface CofferContent {
    Total: number;
    Items: Reward[];
}